iot_config/src/gateway_service.rs

```diff
@@ -40,33 +75,50 @@ impl iot_config::Gateway for GatewayService {
         &self,
         request: Request<GatewayLocationReqV1>,
     ) -> GrpcResult<GatewayLocationResV1> {
-        // Should this rpc be admin-authorized only or should a requesting pubkey
-        // field be added to the request to do basic signature verification, allowing
-        // open access but discourage endpoint abuse?
         let request = request.into_inner();
 
-        let gateway_address: &PublicKeyBinary = &request.gateway.into();
+        let signer = self.verify_public_key(&request.signer)?;
+        self.verify_request_signature(&signer, &request)?;
 
-        let location = self
-            .follower_service
-            .clone()
-            .resolve_gateway_info(gateway_address)
+        let address: &PublicKeyBinary = &request.gateway.into();
+
+        let location = gateway_info::db::get_info(&self.metadata_pool, address)
```

Comment:
NIT: I think you can get location in one step and make it more readable by not using the question mark until the end of the chain.
```rust
        let location = gateway_info::db::get_info(&self.metadata_pool, address)
            .await
            .map_err(|_| Status::internal("error fetching gateway info"))
            .and_then(|opt| {
                opt.ok_or_else(|| {
                    Status::not_found(format!("gateway not found: pubkey = {address:}"))
                })
            })
            .and_then(|iot_metadata| {
                iot_metadata.location.ok_or_else(|| {
                    Status::not_found(format!("gateway unasserted: pubkey = {address:}"))
                })
            })
            .and_then(|location| {
                Cell::from_raw(location).map_err(|_| {
                    Status::internal(format!(
                        "invalid h3 index location {location} for {address}"
                    ))
                })
            })
            .map(|cell| cell.to_string())?;
```